psamm/metabolicmodel.py

```diff
@@ -145,12 +145,34 @@ def __getitem__(self, key):
         return self._create_bounds(key)
 
     def __iter__(self):
-        return self._model.reactions
+        return iter(self._model.reactions)
 
     def __len__(self):
         return sum(1 for _ in self._model.reactions)
 
 
+class _ListWithKey(list):
```

Comment:
Subclassing `list` seems to be discouraged. Better to implement the `collections.Sequence` interface. This also avoids the awkwardness of overriding `__setitem__` to raise a `RuntimeError`.